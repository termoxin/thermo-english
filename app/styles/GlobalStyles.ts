import { createGlobalStyle } from 'styled-components'
import { ThemeType } from 'ui'

const GlobalStyles = createGlobalStyle`
  html,
  body {
    padding: 0;
    margin: 0;
    font-family: 'Rubik', sans-serif;
    font-size: 10px;
  }

  body {
    background: ${({ theme }) => (theme as ThemeType).body.background}
  }

  a {
    color: inherit;
    text-decoration: none;
  }

  * {
    box-sizing: border-box;
  }
`

export default GlobalStyles
