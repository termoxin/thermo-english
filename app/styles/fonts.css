@import url('https://fonts.googleapis.com/css2?family=Amaranth:wght@700&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Rubik&display=swap');
