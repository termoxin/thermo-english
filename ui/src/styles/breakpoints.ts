import {
  css,
  CSSObject,
  FlattenSimpleInterpolation,
  SimpleInterpolation,
} from 'styled-components'

interface BreakpointData {
  value: number
}

type deviceBreakbointsType = keyof typeof deviceBreakpoints

type mediaQuaries = {
  [key in deviceBreakbointsType]: (
    first: TemplateStringsArray | CSSObject,
    ...interpolations: SimpleInterpolation[]
  ) => FlattenSimpleInterpolation
}

interface Breakpoints {
  breakpoints: Record<deviceBreakbointsType, BreakpointData>
  media: mediaQuaries
}

const deviceBreakpoints = {
  mobile: { value: 450 },
  tablet: { value: 600 },
  desktop: { value: 900 },
}

const generateMediaQuariesObject = () => {
  const keys = Object.keys(deviceBreakpoints) as deviceBreakbointsType[]

  const mediaQueries = keys.reduce((acc, val) => {
    acc[val] = (
      first: TemplateStringsArray | CSSObject,
      ...interpolations: SimpleInterpolation[]
    ) => css`
      @media only screen and (max-width: ${deviceBreakpoints[val].value /
        16}em) {
        ${css(first, ...interpolations)}
      }
    `

    return acc
  }, {} as mediaQuaries)

  return mediaQueries
}

export const breakpoints: Breakpoints = {
  breakpoints: deviceBreakpoints,
  media: generateMediaQuariesObject(),
}
