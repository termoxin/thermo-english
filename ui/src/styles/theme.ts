const light = {
  name: 'light',
  body: { background: '#fff' },
  heading: { color: '#130f0f' },
  button: {
    background: {
      red: '#FA4A0C',
      green: '#53B175',
      black: '#000',
    },
    color: {
      primary: '#F6F6F9',
    },
  },
  input: {
    background: '#fff',
    color: '#000',
  },
  card: {
    border: '#bebab3',
  },
  postCard: {
    border: '#FA4A0C',
  },
  logo: {
    color: '#FB4F52',
  },
}

export const themes = {
  light,
  dark: {
    ...light,
    name: 'dark',
    body: { background: '#191919' },
    heading: { color: '#fff' },
  },
}
