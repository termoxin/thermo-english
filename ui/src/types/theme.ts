import { themes } from '../themes'

export type ThemeType = typeof themes['light']
