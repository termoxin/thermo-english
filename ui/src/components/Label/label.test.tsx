import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { themes } from '../../styles/theme'
import { Label } from '.'

describe('Label', () => {
  test('should render published state correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={themes.light}>
        <Label state="published">Published</Label>
      </ThemeProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render draft state correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={themes.light}>
        <Label state="draft">Draft</Label>
      </ThemeProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render requested state correctly', () => {
    const { asFragment } = render(
      <ThemeProvider theme={themes.light}>
        <Label state="requested">Requested</Label>
      </ThemeProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
