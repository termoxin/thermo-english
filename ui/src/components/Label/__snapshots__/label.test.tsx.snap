// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render draft state correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM uEoAW"
  >
    Draft
  </div>
</DocumentFragment>
`;

exports[`Label should render published state correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM dxqALS"
  >
    Published
  </div>
</DocumentFragment>
`;

exports[`Label should render requested state correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnxRM juNxhL"
  >
    Requested
  </div>
</DocumentFragment>
`;
